components/display_drivers/include/display_drivers.hpp

```diff
@@ -32,13 +35,11 @@ typedef std::function<void(int sx, int sy, int ex, int ey, const uint8_t *color_
  * @brief Config structure for all display drivers.
  */
 struct Config {
-  write_fn lcd_write; /**< Function which the display driver uses to write data (blocking) to the
-                         display. */
-  send_lines_fn lcd_send_lines{
-      nullptr}; /**< Function which the display driver uses to send bulk (color) data (non-blocking)
-                   to be written to the display. If not provided, it will default to using the
-                   provided lcd_write (blocking) call. */
-  gpio_num_t reset_pin;        /**< GPIO used for resetting the display. */
+  write_command_fn write_command; /**< Function which the display driver uses to write commands to
+                                     the display. */
+  send_lines_fn lcd_send_lines;   /**< Function which the display driver uses to send bulk (color)
+                     data (non-blocking) to be written to the display. */
+  gpio_num_t reset_pin;           /**< GPIO used for resetting the display. */
```

Comment:
since `reset` and `data_command` can be `GPIO_NUM_NC` (down below), I'd recommend adding default values for both of them to be `GPIO_NUM_NC` in this struct.

Reply:
And mark in the comment that the parameter is optional depending on the display driver in question having those pins broken out / functional.